src/fmelt.c

```diff
@@ -176,33 +178,79 @@ bool is_default_measure(SEXP vec) {
 
 // maybe unlist, then unique, then set_diff.
 SEXP uniq_diff(SEXP int_or_list, int ncol, bool is_measure) {
+  // Protect input list/vector, unlisting if necessary
   SEXP int_vec = PROTECT(isNewList(int_or_list) ? unlist_(int_or_list) : int_or_list);
+  
+  // Check for duplicated elements in the input vector
```

Comment:
please remove comments that are redundant with what is already obvious from the code